xml/vt_qemu_tpm.xml

```diff
@@ -0,0 +1,161 @@
+    This procedure focuses on the <emphasis>socket</emphasis> interface.
+   </para>
+   <procedure>
+    <step>
+     <para>
+      Create a directory <filename>mytpm0</filename> inside the VM directory&mdash;for example, <filename>/var/lib/libvirt/qemu/sle15sp3</filename>&mdash;to store the TPM states:
+     </para>
+ <screen>&prompt.sudo;mkdir /var/lib/libvirt/qemu/sle15sp3/mytpm0</screen>
+    </step>
+    <step>
+     <para>
+      Start <command>swtmp</command>. It will create a socket file&mdash;for example, <filename>swtpm-sock</filename>&mdash;that &qemu; can use:
+     </para>
+ <screen>
+ &prompt.sudo;swtpm socket
+  --tpmstate dir=/var/lib/libvirt/qemu/sle15sp3/mytpm0 \
+  --ctrl type=unixio,path=/var/lib/libvirt/qemu/sle15sp3/mytpm0/swtpm-sock \
+  --log level=20
+ </screen>
+     <tip>
+      <title>TPM version 2.0</title>
+      <para>
+       By default, <command>swtpm</command> starts a TPM version 1.2 emulator and stores its states in the <filename>tpm-00.permall</filename> directory.
+       To create a TPM 2.0 instance, run:
+      </para>
+ <screen>
+ &prompt.sudo;swtpm socket
+  --tpm2
+  --tpmstate dir=/var/lib/libvirt/qemu/sle15sp3/mytpm0 \
+  --ctrl type=unixio,path=/var/lib/libvirt/qemu/sle15sp3/mytpm0/swtpm-sock \
+  --log level=20
+ </screen>
+      <para>
+       TPM 2.0 states will be stored in the <filename>tpm2-00.permall</filename> directory.
+      </para>
+     </tip>
+    </step>
+    <step>
+     <para>
+      Add the following command line parameters to the &qemusystemarch; command:
+     </para>
+<screen>
+&prompt.user;qemu-system-x86_64 \
+[...]
+-chardev socket,id=chrtpm,path=/var/lib/libvirt/qemu/sle15sp3/mytpm0/swtpm-sock \
+-tpmdev emulator,id=tpm0,chardev=chrtpm \
+-device tpm-tis,tpmdev=tpm0
+</screen>
+    </step>
+    <step>
+     <para>
+      Verify that the TPM device is available in the guest by running the following command:
+     </para>
+<screen>&prompt.user;tpm_version
+TPM 1.2 Version Info:
+Chip Version:        1.2.18.158
+Spec Level:          2
+Errata Revision:     3
+TPM Vendor ID:       IBM
+TPM Version:         01010000
+Manufacturer Info:   49424d00
+</screen>
+    </step>
+   </procedure>
+ </sect1>
+ <sect1 xml:id="tpm-libvirt">
+  <title>Using swtpm with &libvirt;</title>
+  <para>
+   To use swtpm with &libvirt;, add the following TPM device to the guest XML specification:
+  </para>
+<screen>
+&lt;devices>
+ &lt;tpm model='tpm-tis'>
+  &lt;backend type='emulator' version='2.0'/>
+ &lt;/tpm>
+&lt;/devices>
+</screen>
+  <para>
+   &libvirt; will start swtpm for the guest automatically, you do not need to start it manually in advance.
+   The corresponding permall file will be created in <filename>/var/lib/libvirt/swtpm/<replaceable>VM_UUID</replaceable></filename>.
+  </para>
+ </sect1>
+ <sect1 xml:id="tpm-ovmf">
+  <title>TPM measurement with OVMF firmware</title>
+  <para>
+   If the guest uses the OVMF firmware, it will measure components with TPM.
```

Comment:
I did not know this abbreviation, so perhaps other readers may not. It stands for Open Virtual Machine Firmware: http://www.linux-kvm.org/downloads/lersek/ovmf-whitepaper-c770f8c.txt

It is an implentation of UEFI for VMs, as opposed to the legacy BIOS. Should we mention this?